app/components/Welcome.js

```diff
@@ -19,25 +32,56 @@ export default class Welcome extends Component {
     setNotFirstTimeFlag()
   }
 
+  selectedLanguage (i) {
+    // set the language to the selected value
+    i18n.changeLanguage(this.state.options[i].code)
+    setDefaultLanguage(this.state.options[i].code)
+  }
+
+  onLanguageChange (event) {
+    // Gets the index of the selected option
+    this.selectedLanguage(event.target.selectedIndex)
+  }
+  renderOptions () {
```

Comment:
Instead of using a separate method, just inline it like settings select box
`            {SUPPORTED_LANGS.map((lang, index) => (
              <option
                key={index}
                index={index}
                value={lang.code}>
                {lang.language}
              </option>
            ))}
`

Reply:
I tried but it didn't worked correctly so i had to do it separately as a workaround, but i'll try again